### PR TITLE
Migrate gcr.io/cloud-marketplace to marketplace.gcr.io.

### PR DESCRIPTION
**Pull request by xingao267**

### CODE REVIEW

**Comment by xingao267** [comment]
@nlopezgi  @nlopezgi @alex1545 Ready for review.

File: configs/dependency-specs/ubuntu16_04.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ revisionsFilePath: "configs/dependency-tracking/ubuntu16_04.yaml"
 gcrDeps:
   # The RBE Ubuntu 16.04 toolchain container.
   - name: "IMAGE"
-    location: "l.gcr.io/google/rbe-ubuntu16-04"
+    location: "marketplace.gcr.io/google/rbe-ubuntu16-04"
```
Comment (nlopezgi) [inline configs/dependency-specs/ubuntu16_04.yaml]: did you double check that all the services / GCB builds that use this work properly when switching from l.gcr.io to marketplace.gcr.io?

File: rules/rbe_repo.bzl [hunk 2]
```diff
@@ -366,7 +365,10 @@ def _use_standard_config(ctx):
 # Pulls an image using 'docker pull'.
 def _pull_image(ctx, image_name):
     print("Pulling image %s." % image_name)
-    result = ctx.execute(["docker", "pull", image_name])
+
+    # Use l.gcr.io registry to pull marketplace.gcr.io images to avoid auth
+    # issues for users who do not do gcloud login.
+    result = ctx.execute(["docker", "pull", image_name.replace("marketplace.gcr.io", "l.gcr.io")])
```
Comment (nlopezgi) [inline rules/rbe_repo.bzl]: are we sure that all containers in marketplace.gcr.io are mirrored in l.gcr.io (i.e., not just rbe_ubuntu)?

File: configs/dependency-specs/ubuntu16_04.yaml (hunk 1 above)
Comment (xingao267) [reply configs/dependency-specs/ubuntu16_04.yaml]: GCB should be able to pull from marketplace.gcr.io. There was issue before, but it has been solved. Will send you the internal thread.

File: rules/rbe_repo.bzl (hunk 2 above)
Comment (xingao267) [reply rules/rbe_repo.bzl]: Yes.

**Pull request merged**